src/pybamm/solvers/processed_variable.py

```diff
@@ -126,22 +493,67 @@ def initialise_0D(self):
                 entries[idx] = float(base_var_casadi(t, y, inputs))
 
                 idx += 1
+        return entries
 
-        if self.cumtrapz_ic is not None:
-            entries = cumulative_trapezoid(
-                entries, self.t_pts, initial=float(self.cumtrapz_ic)
-            )
+    def _observe_postfix(self, entries, _):
+        if self.cumtrapz_ic is None:
+            return entries
 
+        return cumulative_trapezoid(
+            entries, self.t_pts, initial=float(self.cumtrapz_ic)
+        )
+
+    def _interp_setup(self, entries, t):
         # save attributes for interpolation
-        self.entries_for_interp = entries
-        self.coords_for_interp = {"t": self.t_pts}
+        entries_for_interp = entries
+        coords_for_interp = {"t": t}
 
-        self.entries = entries
-        self.dimensions = 0
+        return entries_for_interp, coords_for_interp
+
+    def _shape(self, t):
+        return [len(t)]
+
+
+class ProcessedVariable1D(ProcessedVariable):
+    """
+    An object that can be evaluated at arbitrary (scalars or vectors) t and x, and
+    returns the (interpolated) value of the base variable at that t and x.
+
+    Parameters
+    ----------
+    base_variables : list of :class:`pybamm.Symbol`
+        A list of base variables with a method `evaluate(t,y)`, each entry of which
+        returns the value of that variable for that particular sub-solution.
+        A Solution can be comprised of sub-solutions which are the solutions of
+        different models.
+        Note that this can be any kind of node in the expression tree, not
+        just a :class:`pybamm.Variable`.
+        When evaluated, returns an array of size (m,n)
+    base_variable_casadis : list of :class:`casadi.Function`
```

Comment:
Guessing it was supposed to be this 
```suggestion
    base_variables_casadi : list of :class:`casadi.Function`
```